### ISSUE TITLE
Support for Font Awesome Version 6?

### ISSUE THREAD

**Issue by bdnextiva** [mattmaddux/FASwiftUI#17]
The current version of FA is 6. Will you be upgrading to support FA 6?

**Comment by mattmaddux** [comment]
I haven't had time to try it, honestly. As long as they didn't change the JSON format, it should theoretically still work. Would you give it a try and let me know if it throws any errors?

**Comment by mattmaddux** [comment]
Ah, well that would do it. I've got a number of plans for a V2, and I will add FA V6 support to that list, but I don't have an ETA since my work has taken me away from SwiftUI in the past year and I haven't had the time. I'm open to PRs if anyone wants to contribute.

**Comment by bdnextiva** [comment]
Okay, thanks for the reply. I will see what it will take and consider a PR.

**Comment by bdnextiva** [comment]
There is a JavaScript file that contains the FA 5 to FA 6 mapping that looks like this:

`"wand-magic": [512, 512, ["magic"], "f0d0", "M14.06 395.4L395.4 14.06C414.2-4.686 444.6-4.686 463.3 14.06L497.9 48.64C516.6 67.38 516.6 97.78 497.9 116.5L116.5 497.9C97.78 516.6 67.38 516.6 48.64 497.9L14.06 463.3C-4.686 444.6-4.686 414.2 14.06 395.4V395.4zM354.2 192.3L463.1 82.58L429.4 48L319.7 157.7L354.2 192.3zM285.7 191.7L48 429.4L82.58 463.1L320.3 226.2L285.7 191.7z"],`

I can create a mapping file that is read upon initialization. And in the FontAwesome.swift file and the icon method change the lookup to include the mapping:

```
    public func icon(byName name: String) -> FAIcon? {
        return store[name.lowercased()]
    }
```

Thoughts?

**Comment by bdnextiva** [comment]
We are looking at a more slimed down solution to put a wrapper around something like this:

`                        Text("address-card")
                            .font(Font.custom("FontAwesome6Free-Regular", size: 14, relativeTo: .body))`

**Comment by martyuiop** [comment]
Was able to get FA6 integration working quite quickly by adding 
`case thin`
to the FAStyle enum and by adding a function to add leading zeros to any unicode string less than 4 chars.
(obviously also changing the FACollection enum to 6's instead of 5's and also copy in the FA 6 icons.json and .otf files)

**Comment by mattmaddux** [comment]
That's great. If you open a PR with those changes I will merge it.

**Comment by mattmaddux** [comment]
Thank you! And no rush, because as you can see I'm certainly taking my sweet time with any changes and additions.

**Comment by martyuiop** [comment]
Well that was a fun few hours. Backwards compatibility for 5 / existing deployments done  They changed over 300 icon names and put the old names in an aliases array so this now gets checked for first if a name isn’t found. I’ve submitted the PRIf happy, I’ve recently completed the Kodeco SwiftUI Bootcamp (I have some prior swift and other language stuff too Obv. ) and would be happy to look at being an active contributor. I see the long open issue of the tab items so if you’d be ok to add me as a contributor I’d be happy to take a look at that and submit pr’s appropriately. Likewise if there are any features you want to add, happy to look at that.All the bestMartinOn 17 May 2023, at 16:08, Matt Maddux ***@***.***> wrote:﻿
Thank you! And no rush, because as you can see I'm certainly taking my sweet time with any changes and additions.

**Comment by mattmaddux** [comment]
Thank you for that. I appreciate some action here since it's been a little stagnant. I will take a look at the PR and likely merge it in today.

I might take you up on your offer since I've not been working much in Apple dev lately and thus haven't touched projects like this one.

**Comment by mattmaddux** [comment]
Merged PR #18 to add support for FA 6.
